modules/core/04-channel/keeper/keeper.go

```diff
@@ -653,8 +671,46 @@ func (k Keeper) HasPruningSequenceStart(ctx sdk.Context, portID, channelID strin
 	return store.Has(host.PruningSequenceStartKey(portID, channelID))
 }
 
-// PruneAcknowledgements prunes packet acknowledgements from the store that have a sequence number less than or equal to the pruning sequence.
-// The number of packet acknowledgements pruned is equal to limit. Pruning only occurs after a channel has been upgraded.
-func (Keeper) PruneAcknowledgements(ctx sdk.Context, portID, channelID string, limit, pruningSequence uint64) error {
-	return nil
+// PruneStalePacketData prunes packet acknowledgements and receipts that have a sequence number less than pruning sequence end.
+// The number of packet acks/receipts pruned is equal to limit. Pruning can only occur after a channel has been upgraded.
+//
+// Pruning sequence start keeps track of the packet ack/receipt that can be pruned next. When it reaches pruningSequenceEnd,
+// pruning is complete.
+func (k Keeper) PruneStalePacketData(ctx sdk.Context, portID, channelID string, limit uint64) (uint64, uint64, error) {
```

Comment:
Sorry for all the conflicting names, but PacketData already has a different semantic, its the opaque bytes sent to the application. How about `PruneStalePacketState`?

Reply:
Though honestly was fine with PruneAcknowledgements, since receipts are really just a tiny acknowledgement for core while the full ack is the ack for the app as well